### ISSUE TITLE
App crash on entering data in EditTextField

### ISSUE THREAD

**Issue by longnt06658** [dhis2/dhis2-android-eventcapture#14]
App crash on entering data in EditTextField.
Device: Google Nexus 5
Android Version: 5.1

Log:

java.lang.RuntimeException: Could not dispatch event: class org.hisp.dhis2.android.eventcapture.EditTextValueChangedEvent to handler [EventHandler public void org.hisp.dhis2.android.eventcapture.fragments.dataentry.DataEntryFragment.onRowValueChanged(org.hisp.dhis2.android.eventcapture.EditTextValueChangedEvent)]: Invalid index 6, size is 6
            at com.squareup.otto.Bus.throwRuntimeException(Bus.java:458)
            at com.squareup.otto.Bus.dispatch(Bus.java:388)
            at com.squareup.otto.Bus.dispatchQueuedEvents(Bus.java:369)
            at com.squareup.otto.Bus.post(Bus.java:338)
            at org.hisp.dhis2.android.sdk.utils.MainThreadBus.post(MainThreadBus.java:51)
            at org.hisp.dhis2.android.eventcapture.adapters.rows.dataentry.EditTextRow$OnTextChangeListener.afterTextChanged(EditTextRow.java:166)
            at android.widget.TextView.sendAfterTextChanged(TextView.java:7695)
            at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:9483)
            at android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:972)
            at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:516)
            at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:454)
            at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:33)
            at android.text.method.NumberKeyListener.onKeyDown(NumberKeyListener.java:121)
            at android.widget.TextView.doKeyDown(TextView.java:5803)
            at android.widget.TextView.onKeyDown(TextView.java:5616)
            at android.view.KeyEvent.dispatch(KeyEvent.java:2619)
            at android.view.View.dispatchKeyEvent(View.java:8411)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.widget.ListView.dispatchKeyEvent(ListView.java:2130)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1495)
            at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchKeyEvent(PhoneWindow.java:2361)
            at com.android.internal.policy.impl.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1709)
            at android.app.Activity.dispatchKeyEvent(Activity.java:2702)
            at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchKeyEvent(PhoneWindow.java:2276)
            at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4020)
            at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:3982)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
            at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3680)
            at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
            at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3737)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
            at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5807)
            at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:5781)
            at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:5752)

**Comment by russnes** [comment]
I think this was fixed before the UI refactoring. I think it's related to that when Events are loaded from the server, the API only gives out data values that have been created and have values stored on the server. I used to do a check to see if all DataValues existed and created a new one if some of them were missing.

Edit: It happens for new Events actually. To reproduce enter data in last field in Information Campaign on dev server.
